hopp/simulation/technologies/wind/wind_plant.py

```diff
@@ -191,17 +200,79 @@ def __attrs_post_init__(self):
 
         self._dispatch = None
 
-        self.turb_rating = self.config.turbine_rating_kw
+        if self.config.turbine_rating_kw is not None:
+            self.turb_rating = self.config.turbine_rating_kw
         self.num_turbines = self.config.num_turbines
+            
+        if self.config.model_name=="pysam":
+            self.initialize_pysam_wind_turbine()
+        
+        
+    def initialize_pysam_wind_turbine(self):
+        """Initialize wind turbine parameters for PySAM simulation.
+
+        Raises:
+            ValueError: if invalid turbine name is provided. Print list of valid turbine names before error is raised. 
+            UserWarning: discrepancy in rotor_diameter value
+            UserWarning: discrepancy in hub-height value
+        """
+
+        if self.config.model_name != "pysam":
+            return
+        
         if self.config.rotor_diameter is not None:
             self.rotor_diameter = self.config.rotor_diameter
+        
+        if self.config.turbine_name is not None:
+            valid_name = check_turbine_library_for_turbine(self.config.turbine_name)
+            if not valid_name:
+                print_turbine_name_list()
+                ValueError(f"turbine name {self.config.turbine_name} not found in floris or turbine-models library. Please try an available name.")
```

Comment:
nitpick: could you please set this as a `msg` (like you do below) across multiple lines, then raise the error with that msg?